homeassistant/components/weather/__init__.py

```diff
@@ -112,38 +114,52 @@ class WeatherEntity(Entity):
     _attr_ozone: float | None = None
     _attr_precision: float
     _attr_pressure: float | None = None
+    _attr_pressure_unit: str | None = None
     _attr_state: None = None
     _attr_temperature_unit: str
     _attr_temperature: float | None
     _attr_visibility: float | None = None
+    _attr_visibility_unit: str | None = None
+    _attr_precipitation_unit: str | None = None
     _attr_wind_bearing: float | str | None = None
     _attr_wind_speed: float | None = None
+    _attr_wind_speed_unit: str | None = None
 
     @property
     def temperature(self) -> float | None:
-        """Return the platform temperature."""
+        """Return the platform temperature in native units (i.e. not converted)."""
```

Comment:
For clarity, let's have an elaborate docstring similar to this one also on the other properties.